### ISSUE TITLE
conditionalAssignment rule

### ISSUE THREAD

**Issue by valeriachub** [nicklockwood/SwiftFormat#1538]
Hello developers, 
there are some issues with `conditionalAssignment` rule.

`Version 0.52.6 / Swift 5.9`

It reformats the following code:

```
let text: String
if conditionOne {
  text = "Hello World!"
  doSomeStuffHere()
} else {
   text = "Goodbye!"
}
```
to one with errors:

```
let text: String = if conditionOne {
  "Hello World!" // <---- Expression of type `String` is used
  doSomeStuffHere()
} else { // <---- Non-expression branch of `if` expression may only end with a `throw`
  "Goodbye!"
}
```

Could you please check?

Thank you!

**Comment by nicklockwood** [comment]
Cc @calda

**Comment by calda** [comment]
Will take a look

**Comment by calda** [comment]
Here you go: https://github.com/nicklockwood/SwiftFormat/pull/1539

**Comment by nicklockwood** [comment]
@valeriachub fixed in 0.52.7
